conda_libmamba_solver/solver.py

```diff
@@ -157,27 +165,44 @@ def _channel_to_url_or_name(channel):
 
         return tuple(channels)
 
-    def whoneeds(self, query: str):
-        return self._query.whoneeds(query, self._format)
+    def whoneeds(self, query: str, records=True):
+        result_str = self._query.whoneeds(query, self._format)
+        return self._process_query_result(result_str, records=records)
 
-    def depends(self, query: str):
-        return self._query.depends(query, self._format)
+    def depends(self, query: str, records=True):
+        result_str = self._query.depends(query, self._format)
+        return self._process_query_result(result_str, records=records)
 
-    def search(self, query: str):
-        return self._query.find(query, self._format)
+    def search(self, query: str, records=True):
+        result_str = self._query.find(query, self._format)
+        return self._process_query_result(result_str, records=records)
 
     def explicit_pool(self, specs: Iterable[MatchSpec]) -> Iterable[str]:
         """
         Returns all the package names that (might) depend on the passed specs
         """
         explicit_pool = set()
         for spec in specs:
-            result_str = self.depends(spec.dist_str())
-            result = json_load(result_str)
-            for pkg in result["result"]["pkgs"]:
-                explicit_pool.add(pkg["name"])
+            pkg_records = self.depends(spec.dist_str())
+            for record in pkg_records:
+                explicit_pool.add(record.name)
         return tuple(explicit_pool)
 
+    def _process_query_result(self, result_str, records=True):
+        result = json_load(result_str)
+        if result["result"]["status"] != "OK":
```

Comment:
Should this be written more defensively to catch malformed JSON?

Reply:
Let's see how this looks :)